test/Dialect/FIRRTL/lower-layers-errors.mlir

```diff
@@ -0,0 +1,14 @@
+// RUN: circt-opt -firrtl-lower-layers -split-input-file -verify-diagnostics %s
+
+firrtl.circuit "RWProbeCrossLayer" {
+  firrtl.layer @A bind { }
+  firrtl.module @RWProbeCrossLayer() attributes {layers = [@A]} {
+    %z = firrtl.constant 0 : !firrtl.uint<5>
+    %w = firrtl.node sym @sym %z : !firrtl.uint<5>
+    firrtl.layerblock @A {
+      // expected-error @below {{rwprobe target not moved}}
+      %rw = firrtl.ref.rwprobe <@RWProbeCrossLayer::@sym> : !firrtl.rwprobe<uint<5>>
```

Comment:
This seems more like a verifier error than a `LowerLayers` error.

I.e., this is trying to create a `rwprobe` that can be used to force something outside the layerblock which seems to violate the guarantee that no layerblock operation will "drive" anything outside the layerblock.

WDYT?